client/App/Components/AudioView.js

```diff
@@ -0,0 +1,252 @@
+var React = require('react-native');
+var MapView = require('react-native-maps');
+var Icon = require('react-native-vector-icons/FontAwesome');
+var CircleMarker = require('./CircleMarker');
+var PhotoView = require('./PhotoView');
+var PhotosView = require('./PhotosView');
+var api = require('../Utils/api');
+var BlackPhotoMarker = require('./BlackPhotoMarker');
+var RedPhotoMarker = require('./RedPhotoMarker');
+var IconIon = require('react-native-vector-icons/Ionicons');
+
+var {AudioRecorder, AudioPlayer} = require('react-native-audio');
+
+var {
+  Navigator,
+  StyleSheet,
+  View,
+  Text,
+  Dimensions,
+  TouchableOpacity,
+  TouchableHighlight,
+  StatusBarIOS,
+  ActivityIndicatorIOS,
+  TextInput,
+  Component,
+  DeviceEventEmitter,
+  AlertIOS,
+  AppRegistry
+} = React;
+
+class AudioView extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      latitude: this.props.latitude,
+      longitude: this.props.longitude,
+      userId: this.props.userId || this.props.route.userId,
+
+      recordingNow: false,
+      stoppedRecording: false,
+      stoppedPlaying: false,
+      playing: false,
+      finished: false,
+      currentTime: null,
+      recordingStatusText: null,
+      currentFileName: null
+    };
+  }
+
+  componentDidMount(){
+
+    // upload progress
+    DeviceEventEmitter.addListener('RNUploaderProgress', (data)=>{
+      let bytesWritten = data.totalBytesWritten;
+      let bytesTotal   = data.totalBytesExpectedToWrite;
+      let progress     = data.progress;
+
+      console.log('Bytes Total', bytesTotal);
+      console.log('Bytes Written', bytesWritten);
+
+      console.log( "upload progress: " + progress + "%");
+    });
+
+    AudioRecorder.onProgress = (data) => {
+      console.log('On audio rec progress-------------------');
+      // TODO: parse into nice timestamp here...
+      var parsedTimeStamp = Math.round(data.currentTime);
+      this.state.recordingStatusText = 'Recording...';
+      this.setState({currentTime: parsedTimeStamp });
+    };
+    AudioRecorder.onFinished = (data) => {
+      this.state.recordingStatusText = null;
+      this.state.currentTime = null;
+      this.setState({finished: data.finished});
+      console.log(`Finished recording: ${data.finished}`)
+    };
+
+    setInterval(()=> {
+      if(this.props.params.index===1) {
```

Comment:
This line of code won't return the right locations because of this if statement. The if statement checks which 'page' the app is on based on the swiper in Main.js. Checkout how stanzas are handled on Main.js and in AuthorView to see how to set it up properly